### DISCUSSION TITLE
Workspaces and webpack module federation

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#2765] [Q&A] [answered]
Hi all, I'm not sure if this is more of a yarn issue or webpack issue. I would really appreciate if someone could give me some guidance here as I have no idea what to do. I'm trying to use the module federation plugin with webpack 5 and I'm having an issue when trying to use shared packages when they are yarn 2 workspaces.

The error I'm getting is:
`Module not found: Error: Can't resolve 'workspace:*' in path/to/import/file`

I do have the module installed as  `"@organization/my-module": "workspace:*"`,

Additionally my module federation config for shared looks like this:

```
const deps = require("./package.json").dependencies;

shared: {
    ...deps,
    react: {
        singleton: true,
        requiredVersion: deps.react,
    },
    "react-dom": {
        singleton: true,
        requiredVersion: deps["react-dom"],
    },
    ...sharedWebpackConfig.getSharedPackages()
}

```

I am able to resolve the error by adding 

```
"@organization/my-module": {
  *requiredVersion: *
}
```

to the shared object. However, the issue then becomes that webpack will just load two copies of the package rather than just use one. Can anyone explain what is going on and if there's some way I can make webpack only use one version of a workspace? 

Thanks and I can attach a code example if necessary or file an issue on webpack if this is on their end. I just need some more information.

**Comment by merceyz** [comment, marked as answer]
Would probably be better to file an issue at https://github.com/webpack/webpack

**Reply by wdfinch** [reply]
Ok thanks!
